sinks/splunk/splunk.go

```diff
@@ -46,24 +45,74 @@ func NewSplunkSpanSink(server string, token string, localHostname string, valida
 		client.SetHTTPClient(httpC)
 	}
 
-	return &splunkSpanSink{Client: client, hostname: localHostname, log: log, sendTimeout: sendTimeout}, nil
+	return &splunkSpanSink{
+		Client:      client,
+		ingest:      make(chan *hec.Event),
+		flush:       make(chan []*hec.Event),
+		hostname:    localHostname,
+		log:         log,
+		sendTimeout: sendTimeout,
+	}, nil
 }
 
 func (sss *splunkSpanSink) Start(cl *trace.Client) error {
 	sss.traceClient = cl
+	go sss.batchAndSend()
+
 	return nil
 }
 
+func (sss *splunkSpanSink) batchAndSend() {
+	batch := make([]*hec.Event, 0, 1)
+
+	for {
+		select {
+		case ev := <-sss.ingest:
+			batch = append(batch, ev)
+		case sss.flush <- batch:
+			// We could flush the batch - get us a new one.
+			batch = make([]*hec.Event, 0, len(batch))
+		}
+
+	}
+}
+
+// Flush takes the batched-up events and sends them to the HEC
+// endpoint for ingestion. If set, it uses the send timeout configured
+// for the span batch.
 func (sss *splunkSpanSink) Flush() {
+	flushed := 0
+	dropped := 0
+	batch := <-sss.flush
+
+	// TODO: Ideally, Flush() would get a context of its own:
```

Comment:
The reason we didn't do this originally is that contexts are generally useful for tracing, and if we're flushing once per span, we'd write-amplify. We could change the interface method signature if that's useful, though.

Reply:
I don't follow the reasoning about contexts there - I can write-amplify spans without a context already (-:

A context on the method signature would allow this Flush() to run in parallel with other Flush()es (with a deadline to ensure no sinks get backed up)... I think I'll make a follow-up PR to do that.